clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithCommitFailedExceptionForTimeout() {
```

Comment:
Ah, the best way to make sure that future doesn't complete is not to prepare `offsetCommitResponse`. Sorry I got that bit late. Thank you @hachikuji for pointing that. I tested it with and without the fix and it works as expected. Will be sending the updated PR soon.